src/shared/components/Slider/Slider.test.tsx

```diff
@@ -0,0 +1,111 @@
+import { render, screen, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import Slider from './Slider';
+import { vi, it, describe, expect, beforeEach } from 'vitest';
+
+//UNIT TESTING
+/* 
+- 1 Render a component we test
+- 2 Simulating user moving the slider, calling onChange
+- 3 Handles min and max boundaries correctly - It ensures that even if you try to set a value outside the range, the component's onChange function is still called with a valid number within the range.
+- This test confirms that the slider respects the step prop
+checks for invalid boundary values. Y
+*/
+describe('Slider Component', () => {
+  const mockOnChange = vi.fn();
+  const defaultProps = {
+    min: 0,
+    max: 100,
+    step: 5,
+    value: 50,
+    onChange: mockOnChange,
+  };
+
+  beforeEach(() => {
+    mockOnChange.mockClear();
```

Comment:
prefer to create a new spy on each call over re-using the same one after clearing its state. Clearing can be used in some scenario where we need to re-use the same spy with a clear state but it is something very rare

```suggestion
      let mockOnChange: ReturnType<typeof vi.fn>;

      beforeEach(() => {
        mockOnChange = vi.fn();
      });
```